tests/unit_tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -145,11 +137,25 @@ async def test_new_monitor_can_pick_up_where_we_left_off(evaluator):
                 {"current_memory_usage": 1000},
             )
 
-        evt = await events.__anext__()
-        snapshot = Snapshot(evt.data)
-        assert snapshot.get_job("0", "0").status == FORWARD_MODEL_STATE_RUNNING
-        assert snapshot.get_job("1", "0").status == FORWARD_MODEL_STATE_RUNNING
-        assert snapshot.get_job("1", "1").status == FORWARD_MODEL_STATE_RUNNING
+        final_snapshot = Snapshot({})
+
+        def check_if_all_fm_running(snapshot: Snapshot) -> bool:
+            try:
+                assert snapshot.get_job("0", "0").status == FORWARD_MODEL_STATE_RUNNING
+                assert snapshot.get_job("1", "0").status == FORWARD_MODEL_STATE_RUNNING
+                assert snapshot.get_job("1", "1").status == FORWARD_MODEL_STATE_RUNNING
+                return True
+            except AssertionError:
+                return False
```

Comment:
Isn't the point to actually raise if they all are not RUNNING state?

Reply:
Yes, but when I removed the time.sleep(), we got a snapshot_event then snapshot_update_event. Here I loop over all events; merge them; then check if it the snapshot is complete. I can probably put it into a function and use asyncio.wait_for instead.